File: core/src/main/java/no/elg/infiniteBootleg/console/consoles/StdConsole.java
package no.elg.infiniteBootleg.console.consoles;

import com.strongjoshua.console.HeadlessConsole;
import no.elg.infiniteBootleg.console.HelpfulConsoleHelpUtil;

/** A console that reads input from standard in */
public class StdConsole extends HeadlessConsole {

  @Override
  public void printHelp(String command) {
    HelpfulConsoleHelpUtil.printHelp(this, exec, command);
  }

  @Override
  public void printCommands() {
    HelpfulConsoleHelpUtil.printCommands(this, exec);
  }

  @Override
  public boolean isVisible() {
    return true;
  }
}

File: core/src/main/kotlin/no/elg/infiniteBootleg/console/consoles/StdConsole.kt
package no.elg.infiniteBootleg.console.consoles

import com.strongjoshua.console.HeadlessConsole
import no.elg.infiniteBootleg.console.HelpfulConsoleHelpUtil

/** A console that reads input from standard in  */
class StdConsole : HeadlessConsole() {
  override fun printHelp(command: String) {
    HelpfulConsoleHelpUtil.printHelp(this, exec, command)
  }

  override fun printCommands() {
    HelpfulConsoleHelpUtil.printCommands(this, exec)
  }

  override fun isVisible(): Boolean = true
}
